File: scripts/frontend/use_brute_force_nns.js
import { PointLoader } from "../../src/frontend/utils/points_loader.js";
import { Point } from "../../src/frontend/utils/point.js";
import { BruteForceNNS } from "../../src/frontend/search/brute_force.js";

console.log("Loading points.")
let points = PointLoader.load("./data/sift/siftsmall/siftsmall_base.bin", 128) 
points = points.map((point, index) => new Point(index, point))

console.log("Building indexer.")
const indexer = new BruteForceNNS()
indexer.build(points)

console.log("Querying indexer.") 
const target = points[100] 
const k = 10 
console.log(indexer.nearest(target, k))

File: scripts/backend/use_brute_force_nns.py
from src.backend.utils.points_loader import PointsLoader 
from src.backend.search.brute_force import BruteForceNNS
from src.backend.utils.point import Point

print("Loading points.")
points = PointsLoader.load("./data/sift/siftsmall/siftsmall_base.bin", 128)
points = [Point(index, points[index]) for index in range(len(points))]

print("Building indexer.")
indexer = BruteForceNNS() 
indexer.build(points)

print("Querying indexer.")
target = points[100] 
k      = 10 
print(indexer.nearest(target, 10))
